laterpay/application/Controller/Admin/Pricing.php

```diff
@@ -545,22 +550,28 @@ protected function time_pass_save( LaterPay_Core_Event $event ) {
             $voucher_titles = $save_time_pass_form->get_field_value( 'voucher_title' );
             foreach ( $voucher_codes as $idx => $code ) {
                 // normalize prices and format with 2 digits in form
-                $voucher_price = isset( $voucher_prices[ $idx ] ) ? $voucher_prices[ $idx ] : 0;
+                $voucher_price           = isset( $voucher_prices[ $idx ] ) ? $voucher_prices[ $idx ] : 0;
+                $formatted_voucher_price = number_format( LaterPay_Helper_View::normalize( $voucher_price ), 2, '.', '' );
+
+                if ( floatval( $formatted_voucher_price ) > floatval( $data['price'] ) ) {
+                    $formatted_voucher_price = $data['price'];
+                }
+
                 $vouchers_data[ $code ] = array(
-                    'price' => number_format( LaterPay_Helper_View::normalize( $voucher_price ), 2, '.', '' ),
+                    'price' => $formatted_voucher_price,
                     'title' => isset( $voucher_titles[ $idx ] ) ? $voucher_titles[ $idx ] : '',
                 );
             }
         }
 
         // save vouchers for this pass
-        LaterPay_Helper_Voucher::save_pass_vouchers( $pass_id, $vouchers_data );
+        LaterPay_Helper_Voucher::save_vouchers( $pass_id, $vouchers_data );
 
         $data['category_name']   = get_the_category_by_ID( $data['access_category'] );
         $hmtl_data               = $data;
         $data['price']           = number_format( $data['price'], 2, '.', '' );
         $data['localized_price'] = LaterPay_Helper_View::format_number( $data['price'] );
-        $vouchers                = LaterPay_Helper_Voucher::get_time_pass_vouchers( $pass_id );
+        $vouchers                = LaterPay_Helper_Voucher::get_time_pass_subscription_vouchers( $pass_id );
```

Comment:
Since now vouchers are being used in time pass as well as subscription can we remove those words from all such functions? In this case it should be `::get_vouchers`? @thrijith

Reply:
Or instead of having a common method and passing params we can have two different methods and internally it can handle the param.